### ISSUE TITLE
Some tensorflow problems 

### ISSUE THREAD

**Issue by schuang23** [yasaminjafarian/HDNet_TikTok#5]
Hi yasamin,thanks to your wonderful works.

When I run your Inference code(python HDNet_Inference.py),I get some errors.

Original stack trace for 'hourglass_stack_fused_depth_prediction/conv1/weights/Initializer/truncated_normal/TruncatedNormal':
  File "HDNet_Inference.py", line 62, in <module>
    out2_1 = hourglass_refinement(x1,True)
  File "/home/schuang/rti3_surreal/HDNet_TikTok/hourglass_net_depth.py", line 122, in hourglass_refinement
    out0_d = hourglass_stack_no_incep(netIN)
  File "/home/schuang/rti3_surreal/HDNet_TikTok/hourglass_net_depth.py", line 77, in hourglass_stack_no_incep
    c0 = conv_layer(stack_in,layer_name('conv'),KER_SZ,NUM_CH[0],bn=bn,training=training)
  File "/home/schuang/rti3_surreal/HDNet_TikTok/hourglass_net_depth.py", line 37, in conv_layer
    kernel = variable('weights', [kernel_size, kernel_size, input_channels, output_channels], initializer, regularizer=tf.contrib.layers.l2_regularizer(0.0005))
  File "/home/schuang/rti3_surreal/HDNet_TikTok/hourglass_net_depth.py", line 31, in variable
    return tf.get_variable(name, shape, initializer=initializer, regularizer=regularizer, dtype=tf.float32, trainable=True)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 1500, in get_variable
    aggregation=aggregation)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 1243, in get_variable
    aggregation=aggregation)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 567, in get_variable
    aggregation=aggregation)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 519, in _true_getter
    aggregation=aggregation)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 933, in _get_single_variable
    aggregation=aggregation)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 258, in __call__
    return cls._variable_v1_call(*args, **kwargs)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 219, in _variable_v1_call
    shape=shape)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 197, in <lambda>
    previous_getter = lambda **kwargs: default_variable_creator(None, **kwargs)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 2519, in default_variable_creator
    shape=shape)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 262, in __call__
    return super(VariableMetaclass, cls).__call__(*args, **kwargs)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 1688, in __init__
    shape=shape)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variables.py", line 1818, in _init_from_args
    initial_value(), name="initial_value", dtype=dtype)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/variable_scope.py", line 905, in <lambda>
    partition_info=partition_info)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/contrib/layers/python/layers/initializers.py", line 150, in _initializer
    seed=seed)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/random_ops.py", line 175, in truncated_normal
    shape_tensor, dtype, seed=seed1, seed2=seed2)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/ops/gen_random_ops.py", line 1016, in truncated_normal
    name=name)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/op_def_library.py", line 794, in _apply_op_helper
    op_def=op_def)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3357, in create_op
    attrs, op_def, compute_device)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 3426, in _create_op_internal
    op_def=op_def)
  File "/home/schuang/anaconda3/lib/python3.7/site-packages/tensorflow_core/python/framework/ops.py", line 1748, in __init__
    self._traceback = tf_stack.extract_stack()

I also try adding these lines of code immediately after importing Tensorflow. But there's no effect.

from tensorflow.compat.v1 import ConfigProto
from tensorflow.compat.v1 import InteractiveSession
config = ConfigProto()
config.gpu_options.allow_growth = True
session = InteractiveSession(config=config)

How can I solve this problem?
Thanks a lot!

**Comment by yasaminjafarian** [comment]
Hi. Thanks a lot. Is it an error or warning? If it is a warning there is no problem it will run the code. 
If it is an error, probably one of the requirements are not met. Have you tried installing with docker? The Cuda version of this container is tested before on this code.
`sudo docker run --gpus all -it --rm -v local_dir:container_dir nvcr.io/nvidia/tensorflow:19.02-py3`
Then don't forget to install the requirements
`pip install -r requirements.txt `

**Comment by schuang23** [comment]
Thank you for your answer.
It's an error.

After I try your method ,the code can run in docker Interactive mode.
However when I exit from this mode,the same error still occurs.
Is this method can only run in docker Interactive mode?

And I also wonder is there a pytorch version code?
Thank you!!

**Comment by yasaminjafarian** [comment]
It is probably because of a mismatch between the versions of the libraries as it has been tested outside of docker.
The code is tested under: Python 3.7.4, CUDA 10 (version 10.0.130) and cuDNN 7 (version 7.4.2)

The tested libraries: 
scipy==1.1.0
tensorflow-gpu==1.14.0
gast==0.2.2 

There is no pytorch version yet.
